sdk/src/azure/core/az_mqtt5_rpc.c

```diff
@@ -14,21 +14,11 @@
 
 #include <azure/core/_az_cfg.h>
 
+static const az_span service_group_id_key = AZ_SPAN_LITERAL_FROM_STR("{serviceGroupId}");
```

Comment:
Enhancement: the codec implementation should be separated from the HFSM one. An app-developer should be able to pick the codec .h/.c files (and core dependencies) and use them within their project, without the HFSM portions being included.